### PR TITLE
Remove redundant constraints

### PR DESCRIPTION
**Pull request by Unisay**
- [x] Remove redundant constraints.
- [x] Enable warning about redundant constraints.

### CODE REVIEW

**Comment by Unisay** [comment]
bors try

File: lib/wallet/integration/src/Test/Integration/Framework/DSL.hs [hunk 1]
```diff
@@ -2248,7 +2193,8 @@ selectCoinsWith ctx w payments transform = do
         (Link.selectCoins @style w) Default (transform payload)
 
 delegationFee
-    :: forall w m. (HasType (ApiT WalletId) w, MonadIO m, MonadUnliftIO m)
+    :: forall w m
+     . (HasType (ApiT WalletId) w, MonadUnliftIO m)
```
Comment (jonathanknowles) [inline lib/wallet/integration/src/Test/Integration/Framework/DSL.hs:2197]:
```suggestion
    :: forall w m. (HasType (ApiT WalletId) w, MonadUnliftIO m)
```

**Review by jonathanknowles — APPROVED** [review]
Superb! Thanks for doing this! ❤️

**Review by Anviking — APPROVED** [review]
👍

File: lib/wallet/integration/src/Test/Integration/Framework/DSL.hs (hunk 1 above)
Comment (Unisay) [reply lib/wallet/integration/src/Test/Integration/Framework/DSL.hs:2197]: I actually did it intentionally :) +1 more reason to have an auto-formatter :-)

**Pull request merged**